community/codegen/src/main/java/org/neo4j/codegen/ExpressionTemplate.java

```diff
@@ -219,16 +222,17 @@ protected void templateAccept( CodeBlock method, ExpressionVisitor visitor )
         return expressions;
     }
 
-    static ExpressionTemplate invokeSuperConstructor( final ExpressionTemplate[] parameters )
+    //TODO I am not crazy about the way type parameters are sent here
```

Comment:
Having two arrays that are required to have the same size is kind of smelly and clunky to use. I think we can do something like you propose but it is kind of a big change I think. Furthermore this method is currently only used from one test and not used at all in the production code at all so it is probably not the highest priority to clean up.